sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -360,18 +371,29 @@ void WinHttpTransport::CreateRequestHandle(std::unique_ptr<_detail::HandleManage
   if (m_options.IgnoreUnknownCertificateAuthority)
   {
     auto option = SECURITY_FLAG_IGNORE_UNKNOWN_CA;
-    if (!WinHttpSetOption(
-            handleManager->m_requestHandle, WINHTTP_OPTION_SECURITY_FLAGS, &option, sizeof(option)))
+    if (!WinHttpSetOption(hi.get(), WINHTTP_OPTION_SECURITY_FLAGS, &option, sizeof(option)))
     {
       GetErrorAndThrow("Error while setting ignore unknown server certificate.");
     }
   }
+
+  if (HasWebSocketSupport())
+  {
+    if (!WinHttpSetOption(hi.get(), WINHTTP_OPTION_UPGRADE_TO_WEB_SOCKET, nullptr, 0))
+    {
+      GetErrorAndThrow("Error while Enabling WebSocket upgrade.");
+    }
+  }
```

Comment:
```suggestion
  if (HasWebSocketSupport() &&
    !WinHttpSetOption(hi.get(), WINHTTP_OPTION_UPGRADE_TO_WEB_SOCKET, nullptr, 0))
    {
      GetErrorAndThrow("Error while Enabling WebSocket upgrade.");
    }
```

Reply:
Good idea. Clang-format had a slightly different opinion of this change, but implemented your suggestion.